src/GraphQL/API.hs

```diff
@@ -78,25 +81,31 @@ data Argument (name :: Symbol) (argType :: Type)
 -- https://hackage.haskell.org/package/optional-args-1.0.1)
 data DefaultArgument (name :: Symbol) (argType :: Type)
 
+-- | Convert a type-level 'Symbol' into a GraphQL 'Name'.
+nameFromSymbol :: forall (n :: Symbol) (proxy :: Symbol -> *). KnownSymbol n => proxy n -> Either NameError Name
```

Comment:
if you feel adventurous you could use type application here as well for consistency with the rest of our APIs. I.e.: `nameFromSymbol @name`